### ISSUE TITLE
Access full care plan

### ISSUE THREAD

**Issue by MollieCourtenay** [healthlocker/healthlocker#42]
## Acceptance criteria

+ [x] As a SLaM service user, looking at my goals and tracking page, I can click to load my Care plan summary (pulled from ePJS).

+ [x] As a SLaM service user, I can see my Care Plan summary -simple formatting - see comment below 22nd aug from mol.

+ [x] **If** the date the care plan was written is available in the data we get from Garry: show a timestamp according to that. Assume this is here: mhlSLAMRecoveryCarePlan> Completed_Date

+ [x] **If** the name and role of the clinician who wrote the care plan is available in the data we get from Garry, show a caption detailing that. Assume this is here: mhlSLAMRecoveryCarePlan>  Completed_By

+ [x] **If** the service user doesn't have a care plan summary in the subset DB, then show this:
Healthlocker is unable to pull your care plan from your record, this may be because:
- your care team use a different care plan template
- your care team have uploaded your care plan as a document rather than completing the fields in clinical record system.

Soon there will be a new care plan which all teams can use. Please be patient whilst this is rolled out. 
If you want a copy of your care plan, ask your care coordinator.

**Comment by LauraNevo** [comment]
This is where it lives, in the Toolkit at the bottom of the page.

![care plan](https://cloud.githubusercontent.com/assets/3858131/22545651/809f2c7e-e931-11e6-8a1b-550056076821.png)

I'm not sure in what format we will be able to get it, if we can pull content directly from ePJS or have to use a .pdf, etc... @MollieCourtenay might know more about that.

**Comment by MollieCourtenay** [comment]
The original myhealthlocker pulled in the care plan from epjs as content not a document.

It should exist as a document in ePJS, but wont always be labelled, named well enough for us to reliably locate the latest one.

We can ask Garry in the ePJS team for the full query to pull in care plan for HL.

**Comment by LauraNevo** [comment]
Quick update about the wireframes. Nothing has changed for this user story, it's just to illustrate the design is evolving and features on this page are growing.

![goals coping strategies](https://cloud.githubusercontent.com/assets/3858131/22781302/d24137c4-eeb9-11e6-8c70-248ccd206f04.png)

**Comment by LauraNevo** [comment]
Update on this:

![care plan](https://cloud.githubusercontent.com/assets/3858131/23085416/a0da9084-f55f-11e6-8b32-537dc9ce0e13.png)

**Comment by katbow** [comment]
I'm not adding a label yet for the full feature, however I'm doing the setup for this now, which is T1h 👍

**Comment by iteles** [comment]
For now, we have simply created a link to a page with a placeholder line of text saying the care plan will come soon as we don't know which fields we will be getting yet.

**Comment by MollieCourtenay** [comment]
The only this with this is that this story begins: As a SLaM service user - and this looks like any user will view this button at the moment.

Will this be picked up when the full feature is built? @iteles @katbow

**Comment by markwilliamfirth** [comment]
@katbow would you be able to update us on the status of this issue and the next steps @RobStallion  needs to take here and if there is anything he should be aware of when implementing it?

**Comment by katbow** [comment]
The only thing that's been done for the care plan is the coming soon page for connected users. 

He'll need to know what table it comes from in ePJS & what information to display from it. If I remember correctly, it contains a lot of information & need to know if it's all meant to be displayed to the user.

**Comment by markwilliamfirth** [comment]
Thanks @katbow 

Assigning back to @RobStallion 

Requires prioritisation by @katyedwards and @MollieCourtenay

**Comment by katbow** [comment]
Need to know: where to access care plan (which table) & which fields need to be accessed. Is it all the fields in the relevant table, or just some of them? 

Information would be displayed on a page (not a document download).

**Comment by markwilliamfirth** [comment]
@cleop to take a look into this one

**Comment by RobStallion** [comment]
Is there an example of the information that the care plan should contain? There are quite a number of tables that have data that relates to a `CarePlan` in some way or another (see below) but we are not sure which tables we will need to access and what information we need to take from those tables.

![image](https://user-images.githubusercontent.com/15571853/29566893-63eda758-8743-11e7-97ea-f5772fcb4bd4.png)

**Comment by MollieCourtenay** [comment]
@RobStallion do the tables look like they match this? from myhealthlocker original?

![screencapture-adults-slam-nhs-uk-my-health-careplanandtreatment-aspx-1503408910519](https://user-images.githubusercontent.com/25012688/29568091-2b7c1e6e-8747-11e7-8626-4033415c131e.png)

**Comment by MollieCourtenay** [comment]
That list you added looks pretty good.

**Comment by katbow** [comment]
@MollieCourtenay each of those items in the list @RobStallion posted is a table. For the first 2 tables alone, there's a total of **69** fields. There is more information in the other 8 tables, some of which seems like duplication from the first 2 tables. 

It would be extremely useful if we had information about which of these tables we need to access, and which fields in those tables we need to use. Otherwise we need to go through every single one to see which has the correct information. It would save us a lot of time if we could avoid having to do this.

These are all the table schemas with all the fields: 
[mhl_Tables_Schema.xlsx](https://github.com/healthlocker/healthlocker/files/1242396/mhl_Tables_Schema.xlsx)

**Comment by MollieCourtenay** [comment]
how does pulling in all the yello bits feel?

[mhl_Tables_Schema.xlsx](https://github.com/healthlocker/healthlocker/files/1242508/mhl_Tables_Schema.xlsx)

**Comment by MollieCourtenay** [comment]
**bold the titles of each column**
list the content beneath

**bold the titles of each column**
list the content beneath

**bold the titles of each column**
list the content beneath

**Comment by MollieCourtenay** [comment]
@Cleop 
FYI the xlsx file above i have just re-uploaded so please download afresh this morning.
I have made a couple of changes to the yellow boxes selected. 
Please do get in touch today to discuss.

**Comment by Cleop** [comment]
mhlSLAMRecoveryCarePlan
--
Important_To_Me_Detail
Staying_Well_Detail
Like_To_Be_Different_Detail
Staying_In_Control_Detail
Support_To_Communicate_Detail
Others_Might_Do_Detail
Supporters_Action_To_Take_Detail
Previous_Support_Type_Detail
Preferred_Medication_Detail
Unacceptable_Medication_detail
Prefer_Not_To_Share_With

**Comment by katyedwards** [comment]
Message if no care plan data is found works great!

Unable to test the care plan summary as no access to a connected user with a care plan!

**Comment by MollieCourtenay** [comment]
+ [ ] the if no care plan message  -  needs formatting differently - as per AC above. 

Healthlocker is unable to pull your care plan from your record, this may be because:

- your care team use a different care plan template
- your care team have uploaded your care plan as a document rather than completing the fields in clinical record system.

Soon there will be a new care plan which all teams can use. Please be patient whilst this is rolled out.
If you want a copy of your care plan, ask your care coordinator.

**Comment by katyedwards** [comment]
Blocked by #1050

**Comment by MollieCourtenay** [comment]
@katbow 

A couple of cosmetic things I'd like to note - having just been able to see this for the first time.
- remove underscores - looks like you have done that @katbow 
- remove 'mhl' wherever it appears
- remove timestamp - keep date only
-  and add spaces between words of titles (maybe have a tidy of all of them if something we can do)
- add grey dividing lines between the main sections to break it up a bit
- think about options IF there is no content to display: have a line of nothing so there is a gap OR no not display the title if there is no content?

**Comment by katbow** [comment]
@MollieCourtenay for timestamps - I switched that over already on the other branch 🙂 

For removing mhl & add spaces between words of titles - do you want them all to just be what they are title now minus 'mhl'?

> think about options IF there is no content to display: have a line of nothing so there is a gap OR no not display the title if there is no content?

Right now it would be the title with no content underneath. If it ends up being a lot of empty sections, it may be better to consider taking the titles out.

**Comment by MollieCourtenay** [comment]
@katbow ideally I would edit all the titles so its a bit more friendly and clear. 
If I did it would it conflict with your PR?

**Comment by katbow** [comment]
It would but I can fix conflicts pretty easily. But I can also pretty easily just change those titles as I go along, so I can take care of it.

Is this how you would want them?
**mhlSummaryOfNeed** -> Summary of Need
**mhlSLAMRecoveryCarePlan** -> Recovery Care Plan
**mhlSLAMRecoveryCarePlanContingency** -> Contingency (not sure about this one, but not sure if Recovery Care Plan should be repeated for each of them as it's in all of them)
**mhlSLAMRecoveryFocusedCarePlanDailyActivity** -> Daily Activity
**mhlSLAMRecoveryFocusedCarePlanGettingHelpFromOthersGeneral** -> Getting Help from Others
**mhlSLAMRecoveryFocusedCarePlanGoalsAndAsp** -> Goals and Aspirations
**mhlSLAMRecoveryFocusedCarePlanLifeEventsTriggers** -> Life Events Triggers

**Comment by MollieCourtenay** [comment]
Titles:
**mhlSummaryOfNeed** -> Summary of need
<grey line divider>
**mhlSLAMRecoveryCarePlan** -> Recovery Care Plan
<grey line divider>
**mhlSLAMRecoveryCarePlanContingency** -> Contingency Plan 
<grey line divider>
**mhlSLAMRecoveryFocusedCarePlanDailyActivity** -> Daily Activity Plan
<grey line divider>
**mhlSLAMRecoveryFocusedCarePlanGettingHelpFromOthersGeneral** -> Getting help from others
<grey line divider>
**mhlSLAMRecoveryFocusedCarePlanGoalsAndAsp** -> Goals and aspirations
<grey line divider>
Create or edit [your goals](https://www.healthlocker.uk/goal) based on your care plan.
**mhlSLAMRecoveryFocusedCarePlanLifeEventsTriggers** ->Triggers and coping strategies
Create or edit [your coping strategies](https://www.healthlocker.uk/coping-strategy) based on your care plan.
<grey line divider>
[Write a note](go to messages) to your care team about your Care Plan Summary.

**Comment by MollieCourtenay** [comment]
@katbow I would want to re-write all the headings (as above) and subheadings too.

Can I do it - shall I wait for your PR to be merged by Simon?

**Comment by katbow** [comment]
Simon has merged in the PR so yeah you can do it. If you want you can do the text changes and I can add the link for messages so it goes to the correct spot as it will involve me updating the controller for it. There's something I need to refactor there anyway. 

You can make your changes then let me know when you're done then I can carry on from the same branch.

**Comment by MollieCourtenay** [comment]
@katbow thank you, will get cracking.

**Comment by MollieCourtenay** [comment]
@katbow   is it ok within this txt file to move the order about a bit? and to maybe delete a couple of bits? Will that mess with it in any way?

**Comment by katbow** [comment]
If there are bits that need deleting (e.g. title and the info is unnecessary) just need to ensure the whole section is deleted.

E.g. if you no longer want the subheading/info for Important to Me Detail in the Recovery plan section, then need to delete both these sections:

```
<h4>Important To Me Detail</h4>
<p><%= @recovery_care_plan."Important_To_Me_Detail" %></p>
```

There are some that have a for loop around them. If you want to move the text somewhere else, the for loop needs to go with it or else it will break.
```
      <%= for contingency <- @contingency do %>
        <p><%= contingency."Changes_Detail" %></p>
      <% end %>
```

Shouldn't break anything that way!

**Comment by MollieCourtenay** [comment]
great thanks @katbow

**Comment by MollieCourtenay** [comment]
Note for me: 
I have deleted this bit:  <div class="mv2">
    <h3 class="b">Daily Activity Plan</h3>
      <h4>What do I/or am I going to do each day to stay well and achieve my goals</h4>
      <%= for daily_activity <- @daily_activity do %>
        <p><%= daily_activity."Activity_Detail" %></p>
      <% end %>
      <h4>Times Detail</h4>
      <%= for daily_activity <- @daily_activity do %>
        <p><%= daily_activity."Times_Detail" %></p>
      <% end %>
      <h4>Activity Reason Detail</h4>
      <%= for daily_activity <- @daily_activity do %>
        <p><%= daily_activity."Activity_Reason_Detail" %></p>
      <% end %>
      <h4>Daily Act Agree By Other</h4>
      <%= for daily_activity <- @daily_activity do %>
        <p><%= daily_activity."DailyActAgreeByOther" %></p>
      <% end %>
      <h4>Daily Act Agree By</h4>
      <%= for daily_activity <- @daily_activity do %>
        <p><%= daily_activity."DailyActAgreeBy" %></p>
      <% end %>
           <div class=" mb3 pb3 bb b--hl-grey bw2">
    </div>

**Comment by MollieCourtenay** [comment]
@katbow once you have looked at my PR.
only things it would be amazing to add:

+ [ ] for users with a care plan - at the bottom.
[Write a note](go to messages) to your care team about your Care Plan Summary.

+ [ ] for users without a care plan to add to their message:
[ask your care coordinator](go to messages) --- turn this into a link to messages.

**Comment by katbow** [comment]
@MollieCourtenay Here's a screen shot of one section. I just used the title heading as the text for now.

![image](https://user-images.githubusercontent.com/1287388/29969548-8a96fdf0-8f18-11e7-9b9e-4fba5535b09c.png)

Adding those links to the bottom now 👍

**Comment by MollieCourtenay** [comment]
@katbow I just added a second 'm' to accommodation.  a new pull request.... sorry.
I blame SLaM for spelling the table heading wrong! haha

**Comment by iteles** [comment]
@MollieCourtenay @katyedwards Can this now be closed with the merging of PR #1059?

**Comment by katyedwards** [comment]
@iteles it doesn't seem to have been deployed when I look at the site so can't test.

**Comment by iteles** [comment]
@katyedwards Thanks! Makes sense that this will follow from #1061 then 😊

**Comment by katyedwards** [comment]
Have moved this final test into a new issue #1066
